ast/desugar/Desugar.cc

```diff
@@ -149,6 +149,27 @@ ExpressionPtr desugarBody(DesugarContext dctx, core::LocOffsets loc, unique_ptr<
     return body;
 }
 
+// It's not possible to use an anonymous rest parameter in a block, as it always refers to the forwarded arguments
+// from the method. This function raises an error if the anonymous rest arg is present in a parameter list.
+void checkBlockRestArg(DesugarContext dctx, const MethodDef::ARGS_store &args) {
+    auto it = absl::c_find_if(args, [](const auto &arg) { return isa_tree<RestArg>(arg); });
+    if (it == args.end()) {
+        return;
+    }
+
+    auto &rest = cast_tree_nonnull<RestArg>(*it);
+    if (auto *local = cast_tree<UnresolvedIdent>(rest.expr)) {
+        if (local->name != core::Names::star()) {
+            return;
+        }
+
+        if (auto e = dctx.ctx.beginError(it->loc(), core::errors::Desugar::BlockAnonymousRestParam)) {
+            e.setHeader("Anonymous rest parameter in block args");
+            e.addErrorNote("Naming the rest parameter will ensure you can access it in the block");
```

Comment:
```suggestion
            e.addErrorNote("Naming the rest parameter will ensure you can access it in the block");
            e.replaceWith("Name the rest parameter", dctx.ctx.locAt(it->loc().copyEndWithZeroLength()), "_");
```

I believe that this autocorrect will autocorrect `*` to `*_` and thus silence the error.

You can test autocorrects now with a `.autocorrects.exp` file alongside testdata tests.

Reply:
Ah, nice! I wasn't sure what to use for a name in a possible autocorrect, but `_` is great 👍